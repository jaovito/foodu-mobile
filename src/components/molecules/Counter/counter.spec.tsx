import {fireEvent, render} from '@test-utils';
import {Counter} from '.';

describe('Counter tests', () => {
  let counterNumber = 1;

  it('should be defined', () => {
    const counter = render(
      <Counter
        counter={counterNumber}
        onDecrement={() => counterNumber--}
        onIncrement={() => counterNumber++}
      />,
    );

    expect(counter).toBeDefined();
  });

  it('should be able to increment counter', () => {
    const {getByTestId} = render(
      <Counter
        counter={counterNumber}
        onDecrement={() => counterNumber--}
        onIncrement={() => counterNumber++}
      />,
    );

    fireEvent.press(getByTestId('IncrementButtonID'));

    expect(counterNumber).toBe(2);
  });

  it('should be able to decrement counter', () => {
    const {getByTestId} = render(
      <Counter
        counter={counterNumber}
        onDecrement={() => counterNumber--}
        onIncrement={() => counterNumber++}
      />,
    );

    fireEvent.press(getByTestId('DecrementButtonID'));

    expect(counterNumber).toBe(1);
  });

  it('should match with snapshot', () => {
    const counter = render(
      <Counter
        counter={counterNumber}
        onDecrement={() => counterNumber--}
        onIncrement={() => counterNumber++}
      />,
    ).toJSON();

    expect(counter).toMatchSnapshot();
  });
});
