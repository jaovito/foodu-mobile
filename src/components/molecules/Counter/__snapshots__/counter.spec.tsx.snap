// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter tests should match with snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#FFF1E5",
        "borderRadius": 51,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 8,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 8,
        "width": 255,
      },
    ]
  }
>
  <RNGestureHandlerButton
    activeOpacity={0.8}
    borderless={true}
    collapsable={false}
    delayLongPress={600}
    handlerTag={7}
    handlerType="NativeViewGestureHandler"
    onActiveStateChange={[Function]}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    onPress={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FE8C00",
        "borderRadius": 31,
        "height": 63,
        "justifyContent": "center",
        "opacity": 1,
        "width": 63,
      }
    }
    testID="DecrementButtonID"
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": undefined,
            "fontSize": 12,
          },
          [
            {
              "color": "#FFFFFF",
              "fontSize": 31,
            },
          ],
          {
            "fontFamily": "Feather",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      
    </Text>
  </RNGestureHandlerButton>
  <Text
    style={
      [
        {
          "color": "#0D0D0D",
          "fontFamily": "Poppins_400Regular",
          "fontSize": 31,
        },
        {
          "color": "#0D0D0D",
          "fontSize": 47,
        },
      ]
    }
  >
    1
  </Text>
  <RNGestureHandlerButton
    activeOpacity={0.8}
    borderless={true}
    collapsable={false}
    delayLongPress={600}
    handlerTag={8}
    handlerType="NativeViewGestureHandler"
    onActiveStateChange={[Function]}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    onPress={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FE8C00",
        "borderRadius": 31,
        "height": 63,
        "justifyContent": "center",
        "opacity": 1,
        "width": 63,
      }
    }
    testID="IncrementButtonID"
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": undefined,
            "fontSize": 12,
          },
          [
            {
              "color": "#FFFFFF",
              "fontSize": 31,
            },
          ],
          {
            "fontFamily": "Feather",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      
    </Text>
  </RNGestureHandlerButton>
</View>
`;
